lib/lineage.js

```diff
@@ -83,8 +83,75 @@ const fetchHydratedDoc = id => {
   });
 };
 
+const findRow = (rows, id) => id && rows.find(row => row.id === id);
+
+const collectContactIds = docs => {
+  const ids = [];
+  docs.forEach(doc => {
+    const contactId = doc.contact && doc.contact._id;
+    if (contactId && ids.indexOf(contactId) === -1) {
```

Comment:
I need to find a better resource to find out what's supported in Node versions. Any suggestions?